src/cpp/KnowledgeItem.cpp

```diff
@@ -47,3 +47,17 @@ void KnowledgeItem::save_to(BaseStorageSaver &storage) {
 	storage.save_creation_time(creation_time());
 	storage.save_title(title());
 }
+
+double KnowledgeItem::match(SearchCriteria<std::string, std::time_t> *criteria) {
+    double passed = 0, total = 0;
+    for (const auto &condition: criteria->get("title")) {

```

Comment:
Is there any way to search in all text fields? Current implementation will require adding multiple conditions? E.g. if we want to search for "test" in TextNote, we will need to add two conditions with same function contains("test"), but different fields - "title" and "text"? Also match will not return 1 if only title or only text matches the criteria?